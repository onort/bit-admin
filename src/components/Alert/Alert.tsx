import React from "react"
import cx from "classnames"

import styles from "./Alert.scss"

export type AlertTypes = "error" | "default" | "success"

interface Props {
  className?: string
  message: string
  type?: AlertTypes
}

// TODO: Consider adding an onClose prop and close icon
const Alert: React.SFC<Props> = props => {
  const className = cx(
    styles.container,
    { [styles[`${props.type}`]]: props.type },
    props.className
  )
  return (
    <div className={className}>
      <span className={styles.message}>{props.message}</span>
    </div>
  )
}

Alert.defaultProps = {
  type: "default"
}

export default Alert
