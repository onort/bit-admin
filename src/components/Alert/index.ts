import Alert, { AlertTypes } from "./Alert"

export default Alert

export { AlertTypes }
