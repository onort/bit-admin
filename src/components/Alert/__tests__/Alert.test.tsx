import React from "react"
import { shallow } from "enzyme"

import Alert from "../"

const testMessage = "Test message."
const testClassName = "testClassName"

describe("<Alert />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<Alert message={testMessage} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should have text rendered", () => {
    const wrapper = shallow(<Alert message={testMessage} />)
    expect(wrapper.find(".message").text()).toBe(testMessage)
  })

  it("should render correctly if a type prop is passed", () => {
    const wrapper = shallow(<Alert message={testMessage} type="error" />)
    expect(wrapper.find(".container").hasClass("error")).toBe(true)
  })

  it("should be able to have a custom className", () => {
    const wrapper = shallow(
      <Alert message={testMessage} className={testClassName} />
    )
    expect(wrapper.find(".container").hasClass(testClassName)).toBe(true)
  })
})
