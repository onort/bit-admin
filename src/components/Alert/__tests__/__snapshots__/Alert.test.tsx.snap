// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> should match snapshot 1`] = `
<div
  className="container default"
>
  <span
    className="message"
  >
    Test message.
  </span>
</div>
`;

exports[`<Alert /> should match snapshot 2`] = `"<div class=\\"container default\\"><span class=\\"message\\">Test message.</span></div>"`;
